packages/x-data-grid-pro/src/hooks/features/rows/useGridRowAriaAttributes.tsx

```diff
@@ -0,0 +1,67 @@
+import * as React from 'react';
+import {
+  GridTreeNode,
+  useGridSelector,
+  gridFilteredTopLevelRowCountSelector,
+  gridFilteredChildrenCountLookupSelector,
+  gridExpandedSortedRowTreeLevelPositionLookupSelector,
+  GRID_ROOT_GROUP_ID,
+} from '@mui/x-data-grid';
+import { useGridRowAriaAttributes as useGridRowAriaAttributesCommunity } from '@mui/x-data-grid/internals';
+import { useGridPrivateApiContext } from '../../utils/useGridPrivateApiContext';
+import { useGridRootProps } from '../../utils/useGridRootProps';
+
+export const useGridRowAriaAttributes = (addTreeDataAttributes?: boolean) => {
+  const apiRef = useGridPrivateApiContext();
+  const props = useGridRootProps();
+  const getRowAriaAttributesCommunity = useGridRowAriaAttributesCommunity();
+
+  const filteredTopLevelRowCount = useGridSelector(apiRef, gridFilteredTopLevelRowCountSelector);
+  const filteredChildrenCountLookup = useGridSelector(
+    apiRef,
+    gridFilteredChildrenCountLookupSelector,
+  );
+  const sortedVisibleRowPositionsLookup = useGridSelector(
+    apiRef,
+    gridExpandedSortedRowTreeLevelPositionLookupSelector,
+  );
+
+  const addAttributes =
+    addTreeDataAttributes !== undefined ? addTreeDataAttributes : props.treeData === true;
+
+  return React.useCallback(
+    (rowNode: GridTreeNode, index: number) => {
+      const ariaAttributes = getRowAriaAttributesCommunity(rowNode, index);
+
+      if (!addAttributes || rowNode === null) {
```

Comment:
Nitpick: Would remove the need to define `addAttributes` variable
```suggestion
      if (rowNode === null || (!props.treeData && !addTreeDataAttributes)) {
```

Reply:
Will update it to
`if (rowNode === null || !(props.treeData || addTreeDataAttributes))`